src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.Nls.cs

```diff
@@ -3,52 +3,52 @@
 
 using System.Buffers;
 using System.Diagnostics;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System.Globalization
 {
     internal static partial class Normalization
     {
-        private static unsafe bool NlsIsNormalized(string strInput, NormalizationForm normalizationForm)
+        private static unsafe bool NlsIsNormalized(ReadOnlySpan<char> source, NormalizationForm normalizationForm)
         {
             Debug.Assert(!GlobalizationMode.Invariant);
             Debug.Assert(GlobalizationMode.UseNls);
-            Debug.Assert(strInput != null);
+            Debug.Assert(!source.IsEmpty);
+            Debug.Assert(normalizationForm == NormalizationForm.FormC || normalizationForm == NormalizationForm.FormD || normalizationForm == NormalizationForm.FormKC || normalizationForm == NormalizationForm.FormKD);
+
+            Interop.BOOL result;
+            fixed (char* pInput = source)
+            {
+                result = Interop.Normaliz.IsNormalizedString(normalizationForm, pInput, source.Length);
+            }
 
             // The only way to know if IsNormalizedString failed is through checking the Win32 last error
             // IsNormalizedString pinvoke has SetLastError attribute property which will set the last error
             // to 0 (ERROR_SUCCESS) before executing the calls.
+            CheckLastErrorAndThrowIfFailed(nameof(source));
+
+            return result != Interop.BOOL.FALSE;
+        }
+
+        private static unsafe bool NlsIsNormalized(string strInput, NormalizationForm normalizationForm)
```

Comment:
Do we need this overload? I'd expect that callers could just bind to the ROS overload. If we do need it for some reason, can it just delegate to the span overload?